orion/pkg/tests_integration/class_integration_test.go

```diff
@@ -0,0 +1,284 @@
+	}
+
+	assertClass(t, 3, classes[0])
+	assertClass(t, 4, classes[1])
+	assert.EqualValues(t, 2, len(classes))
+}
+
+// Test: Create 4 Classes and GetClassesByProgramAndSemester()
+func Test_GetClassesByProgramAndSemester(t *testing.T) {
+	resetTable(t, domains.TABLE_CLASSES)
+
+	createAllClasses(t)
+
+	// Call GetClassesByProgramAndSemester()
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/classes/program/program1/semester/2020_spring", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusOK, recorder.Code)
+	var classes []domains.Class
+	if err := json.Unmarshal(recorder.Body.Bytes(), &classes); err != nil {
+		t.Errorf("unexpected error: %v\n", err)
+	}
+
+	assertClass(t, 1, classes[0])
+	assertClass(t, 2, classes[1])
+	assert.EqualValues(t, 2, len(classes))
+}
+
+// Test: Create 1 Class, Update it, GetByClassId()
+func Test_UpdateClass(t *testing.T) {
+	resetTable(t, domains.TABLE_CLASSES)
+
+	// Create 1 Class
+	class1 := createClass(1)
+	body1 := createJsonBody(class1)
+	recorder1 := sendHttpRequest(t, http.MethodPost, "/api/classes/v1/create", body1)
+	assert.EqualValues(t, http.StatusOK, recorder1.Code)
+
+	// Update
+	updatedClass := createClass(2)
+	updatedBody := createJsonBody(updatedClass)
+	recorder2 := sendHttpRequest(t, http.MethodPost, "/api/classes/v1/class/program1_2020_spring_class1", updatedBody)
+	assert.EqualValues(t, http.StatusOK, recorder2.Code)
+
+	// Get
+	recorder3 := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/class/program1_2020_spring_class1", nil)
+	assert.EqualValues(t, http.StatusNotFound, recorder3.Code)
+	recorder4 := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/class/program1_2020_spring_class2", nil)
+	assert.EqualValues(t, http.StatusOK, recorder4.Code)
+
+	// Validate results
+	var class domains.Class
+	if err := json.Unmarshal(recorder4.Body.Bytes(), &class); err != nil {
+		t.Errorf("unexpected error: %v\n", err)
+	}
+	assertClass(t, 2, class)
+}
+
+// Test: Create 1 Class, Delete it, GetByClassId()
+func Test_DeleteClass(t *testing.T) {
+	resetTable(t, domains.TABLE_CLASSES)
+
+	// Create
+	class1 := createClass(1)
+	body1 := createJsonBody(class1)
+	recorder1 := sendHttpRequest(t, http.MethodPost, "/api/classes/v1/create", body1)
+	assert.EqualValues(t, http.StatusOK, recorder1.Code)
+
+	// Delete
+	recorder2 := sendHttpRequest(t, http.MethodDelete, "/api/classes/v1/class/program1_2020_spring_class1", nil)
+	assert.EqualValues(t, http.StatusOK, recorder2.Code)
+
+	// Get
+	recorder3 := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/class/program1_2020_spring_class1", nil)
+	assert.EqualValues(t, http.StatusNotFound, recorder3.Code)
+}
+
+// Helper methods
+func createClass(id int) domains.Class {
+	now := time.Now().UTC()
+	later1 := now.Add(time.Hour * 24 * 30)
+	later2 := now.Add(time.Hour * 24 * 31)
+	later3 := now.Add(time.Hour * 24 * 60)
+	switch id {
+	case 1:
+		return domains.Class{
+			ProgramId:  "program1",
```

Comment:
View my comment before about needing to create a Programs object, Semesters object and locations object before persisting trying to insert a class into db.

Basically here you're trying to insert a class linked to a program with id `program1`. But there hasn't been a program even inserted into the db yet! So that's why these creates are giving you errors.
In addition, every time you reset the tables for the test, you may want to consider clearing the other tables as well.